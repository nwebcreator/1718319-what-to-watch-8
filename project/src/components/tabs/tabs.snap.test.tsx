import { configureMockStore } from '@jedmao/redux-mock-store';
import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { Provider } from 'react-redux';
import { Router } from 'react-router-dom';
import Tabs from './tabs';

const mockStore = configureMockStore();
const history = createMemoryHistory();

describe('Component: Tabs', () => {
  it('should render correctly', () => {
    const film = {
      id: 1,
      name: 'string',
      posterImage: 'string',
      previewImage: 'string',
      backgroundImage: 'string',
      backgroundColor: 'string',
      videoLink: 'string',
      previewVideoLink: 'string',
      description: 'string',
      rating: 10,
      scoresCount: 10,
      director: 'string',
      starring: [],
      runTime: 10,
      genre: 'string',
      released: 2010,
      isFavorite: false,
    };

    const store = mockStore();

    const { container } = render(
      <Provider store={store}>
        <Router history={history}>
          <Tabs film={film} reviews={[]} />
        </Router>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
