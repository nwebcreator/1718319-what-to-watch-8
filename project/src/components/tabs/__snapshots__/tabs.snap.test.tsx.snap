// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Tabs should render correctly 1`] = `
<div>
  <div
    class="film-card__desc"
  >
    <nav
      class="film-nav film-card__nav"
    >
      <ul
        class="film-nav__list"
      >
        <li
          class="film-nav__item film-nav__item--active"
        >
          <a
            class="film-nav__link"
            href="/#overview"
          >
            Overview
          </a>
        </li>
        <li
          class="film-nav__item"
        >
          <a
            class="film-nav__link"
            href="/#details"
          >
            Details
          </a>
        </li>
        <li
          class="film-nav__item"
        >
          <a
            class="film-nav__link"
            href="/#reviews"
          >
            Reviews
          </a>
        </li>
      </ul>
    </nav>
    <div
      class="film-rating"
    >
      <div
        class="film-rating__score"
      >
        10
      </div>
      <p
        class="film-rating__meta"
      >
        <span
          class="film-rating__level"
        >
          Awesome
        </span>
        <span
          class="film-rating__count"
        >
          10
           ratings
        </span>
      </p>
    </div>
    <div
      class="film-card__text"
    >
      <p>
        string
      </p>
      <p
        class="film-card__director"
      >
        <strong>
          Director: 
          string
        </strong>
      </p>
      <p
        class="film-card__starring"
      >
        <strong>
          Starring: 
          
        </strong>
      </p>
    </div>
  </div>
</div>
`;
